cmd/dbg/main.go

```diff
@@ -0,0 +1,174 @@
+
+	if len(args) == 2 && args[1] == "backends" {
+		backends()
+	} else if len(args) == 3 && args[1] == "backends" && args[2] == "list" {
+		backendsList()
+	} else if len(args) == 4 && args[1] == "backends" && args[2] == "get" {
+		backendsGet(args[3])
+	} else if len(args) == 2 && args[1] == "general" {
+		general()
+	} else if len(args) == 2 && args[1] == "conf" {
+		readNginxConf()
+	} else if len(args) == 1 {
+		info()
+	} else {
+		fmt.Println("Unknown command.")
+	}
+}
+
+func info() {
+	fmt.Println(`dbg is a tool for quickly inspecting the state of the nginx instance.
+Subcommands:
+
+- dbg backends             Output the dynamic backend information as JSON.
+- dbg backends list        Just list the names of all the backends.
+- dbg backends get <NAME>  Output the backend information only for the backend that has this name.
+- dbg general              Output the other dynamic information as JSON.
+- dbg conf                 Dump the contents of /etc/nginx/nginx.conf`)
+}
+
+func backends() {
+	//Get the backend information from the nginx instance
+	body, requestErr := makeRequest("http://127.0.0.1:18080/configuration/backends")
+	if requestErr != nil {
+		fmt.Println(requestErr)
+		return
+	}
+
+	//Indent the returned JSON
+	var prettyBuffer bytes.Buffer
+	indentErr := json.Indent(&prettyBuffer, body, "", "  ")
+	if indentErr != nil {
+		fmt.Println(indentErr)
+		return
+	}
+
+	//Print the pretty JSON
+	fmt.Println(string(prettyBuffer.Bytes()))
+}
+
+func backendsList() {
+	body, requestErr := makeRequest("http://127.0.0.1:18080/configuration/backends")
+	if requestErr != nil {
+		fmt.Println(requestErr)
+		return
+	}
+
+	//Read the array of backends from the returned JSON
+	var f interface{}
+	unmarshalErr := json.Unmarshal(body, &f)
+	if unmarshalErr != nil {
+		fmt.Println(unmarshalErr)
+		return
+	}
+	backends := f.([]interface{})
+
+	//Just list the names of the backends
+	for _, backendi := range backends {
+		backend := backendi.(map[string]interface{})
+		fmt.Println(backend["name"].(string))
+	}
+}
+
+func backendsGet(name string) {
+	body, requestErr := makeRequest("http://127.0.0.1:18080/configuration/backends")
+	if requestErr != nil {
+		fmt.Println(requestErr)
+		return
+	}
+
+	//Read the array of backends from the returned JSON
+	var f interface{}
+	unmarshalErr := json.Unmarshal(body, &f)
+	if unmarshalErr != nil {
+		fmt.Println(unmarshalErr)
+		return
+	}
+	backends := f.([]interface{})
+
+	//Search for a backend by name and output its config if found
+	for _, backendi := range backends {
+		backend := backendi.(map[string]interface{})
+		if backend["name"].(string) == name {
+			printed, _ := json.MarshalIndent(backend, "", "  ")
+			fmt.Println(string(printed))
+			return
+		}
+	}
+	fmt.Println("A backend of this name was not found.")
+}
+
+func general() {
+	//Get the other (general) information from the nginx instance
+	body, requestErr := makeRequest("http://127.0.0.1:18080/configuration/general")
```

Comment:
Instead of `http://127.0.0.1:18080/configuration/general` being hardcoded here, it can be a global variable.

Also `18080` should be obtained be user configurable.